### DISCUSSION TITLE
Is BlackBox Matrix Multiplication implemented in GPFlow?

### DISCUSSION THREAD

**Discussion by AdrienCorenflos** [GPflow/GPflow#1667] [Q&A] [answered]
I guess this is one of the big selling points of GPyTorch and would like to know if it has been implemented here too?

https://arxiv.org/abs/1809.11165

**Comment by st--** [comment, marked as answer]
No, it hasn't been implemented in GPflow (and it's unlikely this will change in the future). If you do want the BBMM, you need to use GPyTorch. Though I would still suggest carefully checking whether BBMM is the best approach to your use-case (it may or may not be). :)

**Reply by AdrienCorenflos** [reply]
It's not indeed but I got asked by a reviewer to compare my method with it. As my method is coded as an extension of GPFlow: https://github.com/EEA-sensors/parallel-gps, this would have made comparison easier (and more fair) :D

**Reply by st--** [reply]
I would say it's more fair to use the implementation by the people who developed the method. : )
You might be interested in http://proceedings.mlr.press/v139/artemev21a which does compare against the "exact" (quotes intentional) method by Wang et al (referred to as "Iterative GP" in Artemev et al). The CGLB method proposed in that paper is now part of GPflow (see https://github.com/GPflow/GPflow/pull/1706).
